confgenerator/self_logs.go

```diff
@@ -113,14 +97,71 @@ func generateFluentBitSelfLogsComponents(ctx context.Context) []fluentbit.Compon
 			},
 		},
 	}.Components(ctx, fluentBitSelfLogsTag, "self-logs-severity")...)
-	out = append(out, fluentbit.TranslationComponents(fluentBitSelfLogsTag, "severity", severityKey, true,
-		[]struct{ SrcVal, DestVal string }{
-			{"debug", "DEBUG"},
-			{"error", "ERROR"},
-			{"info", "INFO"},
-			{"warn", "WARNING"},
-		})...,
-	)
+	out = append(out, LoggingProcessorModifyFields{
+		Fields: map[string]*ModifyField{
+			"severity": {
+				MoveFrom: "jsonPayload.severity",
+				MapValues: map[string]string{
+					"error": "ERROR",
+					"warn":  "WARNING",
+					"info":  "INFO",
+					"debug": "DEBUG",
+				},
+				MapValuesExclusive: true,
+			},
+		},
+	}.Components(ctx, fluentBitSelfLogsTag, "mapseverityvalues")...)
+	return out
+}
+
+type selfLogTranslationEntry struct {
+	fieldMatch string
+	regexMatch string
+	message    string
+	code       string
+}
+
+var selfLogTranslationList = []selfLogTranslationEntry{
+	{
+		fieldMatch: "message",
+		regexMatch: `\[error\]\s\[lib\]\sbackend\sfailed`,
+		message:    "Loggging_Pipeline_Stopped_Working-Please_Verify_Your_Ops_Agent_Config.",
```

Comment:
Why is the message like this? Can we have a better message? Are we going to have an external documentation for this issues, where we can point?

Reply:
There is a bug in Fluent-bit https://github.com/fluent/fluent-bit/issues/4286 that doesn't accept white-space in a `Modify` filter `Set` action ( it's already been fixed in the newest versions, we still haven't updated over 2.0.5). So the only way I've found to do a readable message is to use underscores `_`. 

I will work on having a better messages. Yes, we are going to have external documentation that is going to be connected in the Error Reporting side.

Reply:
Update messages in [last commit](https://github.com/GoogleCloudPlatform/ops-agent/pull/1300/commits/5cdf34aebbc82a8ad2c54eb84e36fd2e20838c78) .

Reply:
Decided offline for design of this PR to only set the `URL` and track with b/293910334 the work to improve of the log messages. 

@mtabasko Do you think the current set public doc URL's are the correct ones for this case ?

Reply:
Synced with @mtabasko offline about documentation updates. This discussion item is completed.